diff --git a/rastervision_pytorch_learner/rastervision/pytorch_learner/semantic_segmentation_learner.py b/rastervision_pytorch_learner/rastervision/pytorch_learner/semantic_segmentation_learner.py
index 75963ee1..d759a7f0 100644
--- a/rastervision_pytorch_learner/rastervision/pytorch_learner/semantic_segmentation_learner.py
+++ b/rastervision_pytorch_learner/rastervision/pytorch_learner/semantic_segmentation_learner.py
@@ -143,10 +143,11 @@ class SemanticSegmentationLearner(Learner):
                 Parallel(old_conv, new_conv),
                 # sum the parallel outputs
                 AddTensors())
+        elif input_channels < old_conv.in_channels:
+            model.backbone.conv1 = nn.Conv2d(
+                in_channels=input_channels, **old_conv_args)
         else:
-            raise ConfigError(
-                (f'Fewer input channels ({input_channels}) than what'
-                 f'the pretrained model expects ({old_conv.in_channels})'))
+            raise ConfigError(f'Something went wrong')
 
         return model
 
